### ISSUE TITLE
Notebook File/Cell Sace and Recover Problem

### ISSUE THREAD

**Issue by REIGN12** [microsoft/vscode#139197]
Issue Type: <b>Feature Request</b>

Hello, I am using jupyter notebook in VScode, I found it diffcult to reason about what I can recover in the cell or the file; I suppose the controlling logic is confusing; If you guys can make it clearer, it will be a great thing to use notebook in VScode! Thanks a lot

VS Code version: Code 1.63.1 (fe719cd3e5825bf14e14182fddeb88ee8daf044f, 2021-12-14T02:13:54.292Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

**Comment by rebornix** [comment]
> found it diffcult to reason about what I can recover in the cell or the file

Do you mean undo/redo?

**Comment by REIGN12** [comment]
> > found it diffcult to reason about what I can recover in the cell or the file
> 
> Do you mean undo/redo?

Exactly; I found it really difficult to use like when I accidentally delete the whole cell

**Comment by amunger** [comment]
re-opening since the user provided an answer. Also a suggestion from https://github.com/microsoft/vscode-jupyter/discussions/9846 was just to not put adding the cell in the undo/redo stack.
